### ISSUE TITLE
Add default snippetInterface <async\await> and snippetSyntax for async/await  

### ISSUE THREAD

**Issue by Blynskyniki** [cucumber/cucumber-js#956]
Hey. Add support for async \ await in standard snippets. I would also like to see the arrow functions. Here is an example of what is needed.

```javascript
function mySnippet(snippetInterface) {
  return {
    build: function build(opts) {
      let implementation;
      if (snippetInterface === 'callback') {
        implementation = "done null, 'pending'";
      } else {
        implementation = "'pending'";
      }
      const definitionChoices = opts.generatedExpressions.map(
        (generatedExpression, index) => {
          const prefix = index === 0 ? '' : '# ';
          const allParameterNames = generatedExpression.parameterNames.concat(opts.stepParameterNames);
          if (snippetInterface === 'callback') {
            allParameterNames.push('done');
          }
          const parametersStr = allParameterNames.length > 0 ? `(${allParameterNames.join(', ')}) ` : '';
          return `${opts.functionName}('${generatedExpression.source.replace(/'/g, '\\\'')}', ${parametersStr} => { \n \n\n `; },
      );
      return (
        `${definitionChoices.join('')}});`
      );
    },
  };
}
module.exports = mySnippet;
```

**Comment by aslakhellesoy** [comment]
Arrow functions are discouraged in stepdefs because they don't allow referring to the current world as `this`, thus making it impossible to share state cleanly between steps, or user helper functions in the world.

**Comment by smfontevan** [comment]
@aslakhellesoy humm interesting. I had implemented arrow functions in all our step_def files. Is there any work around instead of reverting back to traditional function syntax ?

**Comment by charlierudolph** [comment]
Can add async/await but not the arrow functions

**Comment by advance512** [comment]
Arrow functions lose their reference to the world object, as they are bound to the `this` value where they were defined. I would recommend not going that route for now.

**Comment by advance512** [comment]
We'd love this here in Codefresh.

**Comment by alon-codefresh** [comment]
We'd love this here in Codefresh.

**Comment by charlierudolph** [comment]
resolved in #1003 and released in 4.0.0
